### DISCUSSION TITLE
Copilot hangs on "fetchCompletions"

### DISCUSSION THREAD

**Discussion by noahlh** [community/community#41232] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

This has been happening a lot lately and I'm completely stumped -- frequently Copilot will simply fail to connect, fail to throw and error, and just hang indefinitely in VSCode.  Here's an example of log output:

```
[INFO] [default] [2022-12-09T22:24:14.847Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
```

It never times out, it never completes, and it never throws an error.  My network connection is stable (I'm able to post this message), but for some reason Copilot is dead in the water.

Does anyone have any suggestions on where I can even start debugging this?

**Comment by noahlh** [comment]
One possible clue -- I setup a proxy to try and investigate network packets (Proxyman), and I'm not seeing any outbound requests at all to `copilot-proxy.githubusercontent.com`.  I AM seeing the telemetry requests succeed to `copilot-telemetry.githubusercontent.com`, so network traffic is working.  But maybe there's an issue with the extension itself establishing an outbound connection?  Spitballing here.

**Reply by noahlh** [reply]
Ok next clue:  The telemetry packets indicate that data is flowing, eg:

POST https://copilot-telemetry.githubusercontent.com/telemetry
```
{
  "itemsReceived": 4,
  "itemsAccepted": 4,
  "errors": []
}
````

But still nothing showing up in VSCode from Copilot itself.  Infinite spinning Copilot icon and no answers on screen.

**Comment by noahlh** [comment]
Update:  I've narrowed the issue down to CloudFlare's WARP client/VPN.  For some reason, it causes requests to the Copilot server to abort.  Disabling the service, while disappointing, returns functionality to Copilot.  I've reported the issue to CloudFlare and will report back when/if it's resolved.

**Reply by jfullstackdev** [reply]
that's a nice observation, yup, it can affect Copilot

so as for me, I'm making sure of those things, even different browsers matter and GitHub extensions installed on those browsers

**Comment by MehdiCHEBBAH** [comment]
Any Updates for this, I'm facing the same issue for a while now, and I noticed that when I re-open VSCode, it works for a couple of minutes before the same issue reproduces.
